pkgs/applications/backup/proxmox-backup-client/default.nix

```diff
@@ -16,38 +16,45 @@
   proxmox-backup-client,
 }:
 
+let
+  version = "3.0.1";
+
+  proxmox-backup_src = fetchgit {
+    url = "git://git.proxmox.com/git/proxmox-backup.git";
+    rev = "v${version}";
+    name = "proxmox-backup";
+    hash = "sha256-a6dPBZBBh//iANXoPmOdgxYO0qNszOYI3QtrjQr4Cxc=";
+  };
+
+  proxmox_src = fetchgit {
+    url = "git://git.proxmox.com/git/proxmox.git";
+    rev = "2a070da0651677411a245f1714895235b1caf584";
+    name = "proxmox";
+    hash = "sha256-WH6oW2MB2yN1Y2zqOuXewI9jHqev/xLcJtb7D1J4aUE=";
+  };
+
+  proxmox-fuse_src = fetchgit {
+    url = "git://git.proxmox.com/git/proxmox-fuse.git";
+    rev = "93099f76b6bbbc8a0bbaca9b459a1ce4dc5e0a79";
+    name = "proxmox-fuse";
+    hash = "sha256-3l0lAZVFQC0MYaqZvB+S+ihb1fTkEgs5i9q463+cbvQ=";
+  };
+
+  proxmox-pxar_src = fetchgit {
+    url = "git://git.proxmox.com/git/pxar.git";
+    rev = "6ad046f9f92b40413f59cc5f4c23d2bafdf141f2";
+    name = "pxar";
+    hash = "sha256-I9kk27oN9BDQpnLDWltjZMrh2yJitCpcD/XAhkmtJUg=";
+  };
+in
+
 rustPlatform.buildRustPackage rec {
   pname = "proxmox-backup-client";
-  version = "3.0.1";
+  inherit version;
 
-  srcs = [
```

Comment:
Why?

Reply:
To remove the `rec` ? But it seems that it's still there.

Reply:
So that the next changed line could work

```
sourceRoot = proxmox-backup_src.name;
```

`rec` can't be removed there, the code below uses `pname`.

Reply:
As I have said before, refactors should not piggyback an unrelated treewide. We can study the case against `rec` later.

Also, I love getting rid of `rec`, however I do not like that `let . . . in`.

Reply:
Well, the only alternative I see to not do `let ... in` is to do `elemAt` over `srcs` when referencing instead, but that would be even more ugly, and less efficient.

Reply:
The main concern remains: treewides should not go over their proposal.

Reply:
I concur with both of you, making a decision is challenging here. Yet, from my years of experience, I've observed that when we say, '_We can revisit this later_,' it often translates to it never actually being addressed.

Reply:
@AndersonTorres I don't understand, can you explain what you expect me to change here?

Reply:
That's a very good compromise!

Reply:
About the `rec` thing:

- only `stdenv.mkDerivation` supports this today
- `let` is elegant but unnatural, because it throws the attributes too far from they are referenced.

For now I will allow this usage of "let against rec".

Nonetheless, **remember removing the `rec`**!

Reply:
> For now I will allow this usage of "let against rec".

I don't disagree with the technical reason but what makes you think you hold the authority to decide that here, you wording just sounds very unfriendly and alternative less.

Reply:
@AndersonTorres

> * each package goes to its own commit;
> 
> * if I'm doing something beyond the purpose of the PR, I explain this (especifically the "why"s and "how"s) in the body of the commit.

Point №2 implemented. The offending `rec` was removed. Commit message was elaborated. This felt kinda busywork-ish, to be honest.

As to point №1, I this AFAIK is just not how treewide changes are usually done. Just do `git log -p --grep=treewide` to confirm.

I would agree that one package per commit could a good idea when you actually change the derivations in question, but this does not, and splitting this patchset into 300 separate commits would just spam the log.

Reply:
> Point №2 implemented. The offending rec was removed. Commit message was elaborated. This felt kinda busywork-ish, to be honest.

I am trying a middle ground here.
On a hand, @drupol is right about how easy is to forget things on the backlog.
On the other, it is important to document the changes done in a large PR, especially in this case when a cosmetic refactor piggybacks on another one.

> I would agree that one package per commit could a good idea when you actually change the derivations in question, but this does not, and splitting this patchset into 300 separate commits would just spam the log.

Point taken.

I was thinking on the possibilities of a future bisection, and a treewide monolitic PR makes it a bit problematic.

But treewides are not so hard to debug (except the Elisp ones!).

Reply:
> but what makes you think you hold the authority to decide that yourself

I don't like the tone of your "having authority" as coercitive here.

First, because I'm not a BDFL, so any privileges I have as committer can be revoked potentially at any time.

Second, almost anything I can do is not hard to be undone by other person.

Third, what type of authority you are talking about? You said you agreed on the technical reasons, and I was arguing on technical reasons all along.

Fourth, English is not my native language. I just meant to say I will pass over this and not to pursue the merits and demerits of this particular issue, jumping to the next.

Reply:
> I don't like the tone of your "having authority" as coercitive here.

I said it this way because of the phrasing "For now I will allow this ...", I didn't mean to attack you, I just wanted to tell you that this comes of unfriendly.

> Third, what type of authority you are talking about? You said you agreed on the technical reasons, and I was arguing on technical reasons all along.

As I said above the way you phrased it let it sound like that, but it's fine, no worries. 

> Fourth, English is not my native language. I just meant to say I will pass over this and not to pursue the merits and demerits of this particular issue, jumping to the next.

Mine neither. 
No hard feelings. 

I would suggest moving this conversation to matrix if you want further discussion to not spam everyone else here with the noise.